### ISSUE TITLE
Integration tuya+gsm alarm

### ISSUE THREAD

**Issue by Shaman2** [rospogrigio/localtuya#119]
I have some alarms from aliexpress. They work with tuya apps and have gsm channels. What need for support this alarms?

This alarms have dps from 1 to 40, but need change only dps 1. It set and get string value (for status of alarm): disarmed, armed, home, stay. 

Can localtuya support it?

**Comment by postlund** [comment]
I'm not sure how we would connect to such a device. We need IP connectivity directly to it and I assume these devices are designed to just phone home to the cloud.

**Comment by Shaman2** [comment]
No, thay have local IP-address and work fully as tuya device. Different only in dps, because thay have string value in dps.

I can help with test it

**Comment by Shaman2** [comment]
gsm channel work as additional message

**Comment by Shaman2** [comment]
dps:

1  disarmed
2  20
3  1
4  false
9  true
10 false
15 true
16 100
17 true
18 AQUB...
19 AQUBAQEK...
20 true
21 true
22 1
23 1
24 Normal
25 true
26 BCE....
27 true
28 0
29 true
32 normal
33 
34 false
35 false
36 3
37 0
38 AgA...
39 0
40 3

so for change status on alarm need change dps 1 only. Another dps change settings

**Comment by carlosvroque** [comment]
Nice, I also have a tuya alarm/rf gateway that has a few sensors connected to it, I'll check if it's similar.

**Comment by postlund** [comment]
Sounds reasonable enough to implement. We just need to figure out which platform to use. Do you have any idea which values DP 1 can have?

**Comment by Shaman2** [comment]
> Sounds reasonable enough to implement. We just need to figure out which platform to use. Do you have any idea which values DP 1 can have?

In my case 4 values: armed, disarmes, stay, home

**Comment by postlund** [comment]
Sounds very much like `alarm_control_panel`:

https://github.com/home-assistant/core/blob/dev/homeassistant/components/alarm_control_panel/__init__.py
https://github.com/home-assistant/core/blob/b45ac5159dfaec36a7f832a7b136b16b888cafc2/homeassistant/const.py#L246

Do you know what stay and home means/does?

**Comment by Shaman2** [comment]
> Do you know what stay and home means/does?

stay (home) means that door is armed, but motion sensor at home is not armed

**Comment by carlosvroque** [comment]
I went ahead and checked my device, turns out it's quite different:
![image](https://user-images.githubusercontent.com/65680394/98363321-8ac8a100-200d-11eb-8ae8-6b4f82bb5b7a.png)


101: 1= armed, 2 = disarmed, 3 = home armed, 4 = emergency mode
102: Arm delay
103: Alarm delay
104: Admin password
105: Guest password
107: ??
108: ??
113: ??

I believe the last 3 are the sensors I have connected to the gateway, but unfortunately they never change status and the tuya cli wizard gives me different ID and Key for each sensor.

 I'll do some extra tests by unpairing and pairing and see if anything changes.

**Comment by atv2016** [comment]
I have a digoo hamb I can help with testing this.

**Comment by vampywiz17** [comment]
Hello guys,

if someone has some time, please test this:

https://github.com/vampywiz17/tuyaiot2mqtt

i never tested it with Tuya alarm, so it would be good some helping hand :)
